### PR TITLE
[benchmark] UnsafeRawBufferPointer._copyContents

### PR DESCRIPTION
**Pull request by glessard**
Add a benchmark to measure `UnsafeRawBufferPointer._copyContents`. Prompted by the observation that creating an `[UInt8]` from an `UnsafeRawBufferPointer` is slow, while creating a `[UInt8]` from an `UnsafeBufferPointer<UInt8>` is much faster. This slowness was reported in https://bugs.swift.org/browse/SR-9604.
The fast path to initialize `Array` from a `Collection` ultimately relies on said collection's implementation of `_copyContents`, therefore we benchmark `URBP`'s implementation.

### CODE REVIEW

**Comment by glessard** [comment]
@swift-ci please benchmark

**Comment by glessard** [comment]
@swift-ci please benchmark

**Comment by glessard** [comment]
@swift-ci please smoke test

**Comment by glessard** [comment]
@swift-ci please smoke check

File: benchmark/single-source/BufferFill.swift [hunk 1]
```diff
@@ -119,3 +124,41 @@ public func rawBufferInitializeMemoryExecute(n: Int) {
...
   CheckResults(value == a[r])
 }
+
+var r8: UnsafeRawBufferPointer = .init(start: nil, count: 0)
+var b8: UnsafeMutableBufferPointer<UInt8> = .init(start: nil, count: 0)
+
+public func rawBufferCopyContentsSetup() {
+  assert(r8.baseAddress == nil)
+  let count = a.count * MemoryLayout<Int>.stride
+  let rb = UnsafeMutableRawBufferPointer.allocate(byteCount: count, alignment: 8)
```
Comment (lorentey) [inline benchmark/single-source/BufferFill.swift:134]:
```suggestion
  let rb = UnsafeMutableRawBufferPointer.allocate(
    byteCount: count, 
    alignment: MemoryLayout<Int>.alignment)
```

**Review by lorentey — APPROVED** [review]
🚢

File: benchmark/single-source/BufferFill.swift (hunk 1 above)
Comment (glessard) [reply benchmark/single-source/BufferFill.swift:134]: Correct you are.

**Comment by glessard** [comment]
@swift-ci please smoke test and merge

**Pull request merged**